src/import/importConfiguration.ts

```diff
@@ -0,0 +1,713 @@
+  details: YAMLConfigurationPage,
+  property: string,
+  file: string
+): ImplementationGuideDefinitionPage {
+  const page: ImplementationGuideDefinitionPage = { nameUrl };
+  if (details?.title) {
+    page.title = details.title;
+  }
+  if (details?.generation) {
+    page.generation = parseCodeWithRequiredValues(
+      details.generation,
+      ['html', 'markdown', 'xml', 'generated'],
+      `${property}.generation`,
+      file
+    );
+  }
+  if (details != null) {
+    Object.entries(details).forEach(([key, value]) => {
+      if (key == 'title' || key == 'generation') {
+        return;
+      }
+      if (page.page == null) {
+        page.page = [];
+      }
+      page.page.push(parsePage(key, value as YAMLConfigurationPage, `${property}[${key}]`, file));
+    });
+  }
+  return page;
+}
+
+function parseParameters(yamlConfig: YAMLConfiguration): ImplementationGuideDefinitionParameter[] {
+  const parameters: ImplementationGuideDefinitionParameter[] = [];
+  if (yamlConfig.copyrightYear || yamlConfig.copyrightyear) {
+    parameters.push({
+      code: 'copyrightyear',
+      value: `${yamlConfig.copyrightYear ?? yamlConfig.copyrightyear}`
+    });
+  }
+  if (yamlConfig.releaseLabel || yamlConfig.releaselabel) {
+    parameters.push({
+      code: 'releaselabel',
+      value: `${yamlConfig.releaseLabel ?? yamlConfig.releaselabel}`
+    });
+  }
+  if (yamlConfig.parameters) {
+    for (const [code, values] of Object.entries(yamlConfig.parameters)) {
+      normalizeToArray(values).forEach(value => parameters.push({ code, value: `${value}` }));
+    }
+  } else if (parameters.length === 0) {
+    return; // return undefined rather than an empty []
+  }
+  return parameters;
+}
+
+function parseTemplates(
+  yamlTemplates: ImplementationGuideDefinitionTemplate | ImplementationGuideDefinitionTemplate[],
+  file: string
+): ImplementationGuideDefinitionTemplate[] {
+  return normalizeToArray(yamlTemplates)?.map(t => {
+    const template: ImplementationGuideDefinitionTemplate = {
+      ...t,
+      code: parseSimpleCode(required(t.code, 'templates.code', file), 'templates.code', file),
+      source: required(t.source, 'templates.source', file)
+    };
+    removeUndefinedValues(template);
+    return template;
+  });
+}
+
+function parseMenu(yamlMenu: YAMLConfigurationMenuTree): ConfigurationMenuItem[] {
+  if (yamlMenu == null) {
+    return;
+  }
+  return Object.entries(yamlMenu).map(([name, value]) => {
+    const item: ConfigurationMenuItem = { name };
+    if (typeof value === 'string') {
+      item.url = value;
+    } else {
+      item.subMenu = parseMenu(value);
+    }
+    return item;
+  });
+}
+
+function parseHistory(yamlConfig: YAMLConfiguration, file: string): ConfigurationHistory {
+  const yamlHistory = yamlConfig['history'];
+  if (yamlHistory == null) {
+    return;
+  }
+  const history: ConfigurationHistory = {
+    'package-id': yamlHistory['package-id'] ?? yamlConfig.packageId ?? yamlConfig.id,
```

Comment:
No, because we don't actually change `yamlConfig` at all.  The output `Configuration`'s `packageId` may be defaulted to `id` -- but this function is getting called before the output configuration has been fully instantiated and assigned, so I don't think we can reference it.